File: backend/src/main/java/com/programmers/pcquotation/domain/seller/repository/SellerRepository.java
package com.programmers.pcquotation.domain.seller.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.programmers.pcquotation.domain.seller.entitiy.Seller;

@Repository
public interface SellerRepository extends JpaRepository<Seller, Long> {
	Optional<Seller> findByUsername(String username);

	Optional<Seller> findByApiKey(String apiKey);

	Optional<Seller> findByEmail(String email);
}

File: backend/src/main/java/com/programmers/pcquotation/domain/seller/repository/SellerRepository.kt
package com.programmers.pcquotation.domain.seller.repository

import com.programmers.pcquotation.domain.seller.entity.Seller
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface SellerRepository : JpaRepository<Seller, Long> {
    fun findByUsername(username: String): Seller?

    fun findByApiKey(apiKey: String): Seller?

    fun findByEmail(email: String): Seller?
}